service/snap.go

```diff
@@ -183,23 +182,33 @@ func (s *service) IsSnapshotLicensed(ctx context.Context, symID string, pmaxClie
 	}
 	mutex.Lock()
 	defer mutex.Unlock()
-	if licenseCached == false {
-		symmRepCapabilities, err = pmaxClient.GetReplicationCapabilities(ctx)
+
+	symmRepCapability, ok := symmRepCapabilities[symID]
+	if !ok {
+		repCapabilities, err := pmaxClient.GetReplicationCapabilities(ctx)
 		if err != nil {
 			return err
 		}
-		licenseCached = true
-		log.Infof("License information with Powermax %s is cached", symID)
-	}
-	for i := range symmRepCapabilities.SymmetrixCapability {
-		if symmRepCapabilities.SymmetrixCapability[i].SymmetrixID == symID {
-			if symmRepCapabilities.SymmetrixCapability[i].SnapVxCapable {
-				return nil
+
+		for _, symmCapability := range repCapabilities.SymmetrixCapability {
+			if symmCapability.SymmetrixID == symID {
+				symmRepCapabilities[symID] = symmCapability
+				log.Infof("License information with PowerMax %s is cached", symID)
+				break
 			}
-			return fmt.Errorf("PowerMax array (%s) doesn't have Snapshot license", symID)
+		}
+
+		symmRepCapability, ok = symmRepCapabilities[symID]
+		if !ok {
+			return fmt.Errorf("PowerMax array (%s) is not being managed by Unisphere", symID)
 		}
 	}
-	return fmt.Errorf("PowerMax array (%s) is not being managed by Unisphere", symID)
+
+	if symmRepCapability.SnapVxCapable {
+		return nil
+	}
+
+	return fmt.Errorf("PowerMax array (%s) doesn't have Snapshot license", symID)
```

Comment:
The comment for this function does not reflect the return types. Not related to your changes but an odd function. It does not return a boolean.